### ISSUE TITLE
Helpers bind to nested properties differently in stache/mustache

### ISSUE THREAD

**Issue by akagomez** [bitovi/canjs#1651]
Test: https://github.com/bitovi/canjs/commit/68d438b1c62b8491baf95b8b7e1bc6cde30fe58b

**Comment by kylegifford** [comment]
To clarify the problem a bit:

Binding via dot syntax (`this.attr('parent.child')`):
- works with `this.attr('parent.child', 'foo')`
- does not work with `this.attr('parent', {'child': 'foo'})`

Binding via nested attrs (`this.attr('parent').attr('child')`):
- works with `this.attr('parent', {'child': 'foo'})`
- does not work with `this.attr('parent.child', 'foo')`

**Comment by akagomez** [comment]
@kylegifford I think my fix here in #1655 will at least make the two tests consistent.

**Comment by akagomez** [comment]
I don't think binding to...

```
var map = new can.Map({
  parent: null
})
```

With... 

```
map.attr('parent.child')
```

And expecting that to work after.... 

```
map.attr({ parent: { child: 'foo' }}); 
```

Is officially supported. Although, I certainly wish it was. Either way I think it's a separate issue.

**Comment by kylegifford** [comment]
After applying the fix in #1655 the tests work consistently between Stache and Mustache, and binding via nested attrs now works correctly in Stache. Both still fail using dot syntax mentioned above.

@akagomez can this be closed (after cleaning up tests) since this is working consistently in both Stache and Mustache?

**Comment by akagomez** [comment]
That makes sense to me. I'll create a new issue for the dot syntax binding.

**Comment by kylegifford** [comment]
Tests added to #1655 .

**Comment by daffl** [comment]
Closed via #1655
